doc/api/modules.md

```diff
@@ -481,81 +481,126 @@ added: v0.1.16
 
 * {Object}
 
-The `module.exports` object is created by the Module system. Sometimes this is
-not acceptable; many want their module to be an instance of some class. To do
-this, assign the desired export object to `module.exports`. Note that assigning
-the desired object to `exports` will simply rebind the local `exports` variable,
-which is probably not what you want to do.
+The `module.exports` object is used to export values in a js file. It is
+initialized as `{}` by Module system, and passed to
+[module wrapper](#modules_the_module_wrapper) as argument. (This makes
+`module.exports` and `exports` available in a js file, the details will be
+described below.) `require` returns the processed `module.exports` as the
+exports of a module.
 
-For example suppose we were making a module called `a.js`
-
-```js
-const EventEmitter = require('events');
-
-module.exports = new EventEmitter();
+#### exports alias
+<!-- YAML
+added: v0.1.16
+-->
 
-// Do some work, and after some time emit
-// the 'ready' event from the module itself.
-setTimeout(() => {
-  module.exports.emit('ready');
-}, 1000);
-```
+There is no magic behind `exports`. It is the `module.exports` passed to
+[module wrapper](#modules_the_module_wrapper) as the first argument.
 
-Then in another file we could do
+Here is a minimized code logic of `require` to illustrate the export
+principle, and how `module.exports` and `exports` works:<br/>
 
+As described in [module wrapper](#modules_the_module_wrapper), your js file
+will be wrapped like this:
 ```js
-const a = require('./a');
-a.on('ready', () => {
-  console.log('module a is ready');
+(function (exports, require, module, __filename, __dirname) { // fileWrapper
+// Your module code actually lives in here
+// So both of the exports and module.exports are valid in your file
 });
 ```
+And this is a minimized logic of `require` for `module.exports`:
+```js
+function _require(fileWrapper) {
+  // module.exports is initialized as {}
+  const module = { exports:{} };
+  // The wrapped file will be called with this = module.exports
+  // module.exports is passed as exports. ( exports = module.exports )
+  // module is passed as module.
+  fileWrapper.apply(module.exports,
+    [module.exports, require, module, __filename, __dirname]);
+  // the return value of require, its the export results of a required module.
+  return module.exports;
+}
+```
+When you `require` a file, Node.js will wrap it with a function wrapper as
```

Comment:
I don't think this is a useful way to describe require. If you want to read the code to understand the behaviour, the code is there, its open source. Docs should describe behaviour and interface, not just reformat the code and expect the reader to read the code to figure out how it behaves.